mne/decoding/gsoc.py

```diff
@@ -0,0 +1,523 @@
+# Authors: Jean-Remi King <jeanremi.king@gmail.com>
+#          Asish Panda <asishrocks95@gmail.com>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+import warnings
+from scipy import linalg
+
+from .mixin import TransformerMixin
+from ..epochs import _BaseEpochs
+from ..import Covariance, pick_types
+from ..cov import _regularized_covariance
+from ..preprocessing import Xdawn
+
+
+class _EpochsTransformerMixin(TransformerMixin):
+    """Mixin class for reshaping data to Epoch's standard shape
+
+    This class is meant to be inherited by transformers that are to be
+    used in scikit-learn pipeline. It provides functionality to convert
+    data matrix into 3D.
+
+    Parameters
+    ----------
+    n_chan : int (default : None)
```

Comment:
I think this doc annotation pattern is not common. You would just say in the doc string that it is None